MoTrPAC/motrpac-frontend
# Connects to #106. Google Analytics tag update.

### Key Changes

1. Updated Google Analytics implementation by replacing the use of `analytics.js` with `gtag.js`
2. Removed the no longer needed resource files associated with prior implementation of human meta-analysis
3. Upgraded `react-scripts` dependency to latest version to address issue in which ESLint errors prevent app  from compiling in `development` environment, as well as a vulnerability issue associated with its peer dependency `immer`
4. Removed some ESLint rule overrides that were previously added to app compiling issue caused by ESLint errors
5. Upgraded `storybook` dependencies to address a vulnerability issue associated with its peer dependency `immer`
6. Upgraded a few other outdated dependencies

### Technical Note

Use `nvm` to set `node.js` to `v14.15.4`

### Steps to test

1. `git clone` the repo and run `yarn install` at the repo's root directory.
2. At the repo's root directory, create a `.env` file and add `ESLINT_NO_DEV_ERRORS=true` in the file (for references, see https://github.com/facebook/create-react-app/blob/master/docusaurus/docs/advanced-configuration.md and https://github.com/facebook/create-react-app/pull/10590).
3. Run `yarn inspect-branch`, and expect to see ESLint warnings in the console (both in Terminal and browser) while the app can be successfully compiled and run.
4. Now, run `yarn audit` and expect to see `0 vulnerabilities found`  in the output message.
5. Next, open the frontend project in VS Code. Open some `*.jsx` files (e.g. `releaseEntry.jsx`) and expect to see some lines of code underlined (red _wavy_ lines) - an indication that `eslint` and `prettier` are hinting fixes.
6. Log into MoTrPAC's **Google Analytics** account and select the `dev` property.
7. From its left-hand navbar,  select `Realtime > Content`. Then try clicking through different pages on the local instance of the frontend app. Expect to see the pageviews being reported in real time in Google Analytics.
8. From its left-hand navbar,  select `Realtime > Events`. On the homepage of the local instance of the frontend app, try clicking on the 2 links. Expect to see the events being reported in real time in Google Analytics.
9. Sign into the frontend app and go to the **Releases** page. Try downloading some files (hint: zipped phenotype files are smallest to download). Expect to see the events being reported in real time in Google Analytics.
